example/output.js

```diff
@@ -34,16 +34,16 @@ var main = function(config)
   }
   self.config.direction || (self.config.direction = gpio.DIRECTION.OUT);
   console.log("log: pin" + self.config.pin + ": opening: direction: " + self.config.direction);
-  self.port = gpio.export(self.config.pin, {
-    direction: self.config.direction,
-    ready: function() {
-      console.log("log: pin" + self.config.pin + ": ready:");
-      var intervalTimer = setInterval(function() {
-        self.value = !self.value;
-        self.port.set(self.value);
-        console.log("log: pin" + self.config.pin + ": change: " + self.value);
-      }, 1000);
-    }});
+  self.port = gpio.open(self.config, function(err, gpioPin) {
+    if(err) return console.log(err);
+
+    console.log("log: pin" + self.config.pin + ": ready:");
+    var intervalTimer = setInterval(function() {
+      self.value = !self.value;
+      gpioPin.set(self.value);
```

Comment:
isnt overlapping with self.port ? JS can be confusing here

Reply:
self.port and `gpio.open` callback's second parameter is the same. Via Samsung IoT API, I think it's better to promote using the variable given by the callback if we're writing code within the same callback function.

Reply:
Actually on a related note, I've been thinking `gpio.open` should support an array of config objects in order to export multiple ports and have an elegant callback that ensures all those ports are available to use. I'll make a separate issue for it.

Contract should look something like...

```js
gpio.open([
  { pin: 4, direction: gpio.DIRECTION.IN },
  { pin: 17, direction: gpio.DIRECTION.OUT }
], function(err, gpioPins) {
  if(err) return console.log(err);

  var gpioPin4 = gpioPins[0];
  var gpioPin17 = gpioPins[1];
});
```

Reply:
I guess  if array is not passed as arg , we can assume it's a single gpio pin, right ?

How would you threat the error object ? if !=null then at least one pin failed to be opened ?


BTW, Great minds think alike:

https://github.com/rzr/webthing-iotjs/blob/master/example/platform/board/artik530.js#L31

Reply:
There's actually already an established pattern for this, Promises.all is the first that comes to my head. If any of the asynchronous operations fail, the error object represents the first error that came back. It makes sense since any error/exception should stop code execution, we're only interested in the first error we run into.